### DISCUSSION TITLE
How to customize confirmation dialog to default to `Y/n` instead of `y/N`?

### DISCUSSION THREAD

**Discussion by fritzrehde** [sxyazi/yazi#545] [1. Q&A]
Hi, is this possible? I often delete files permanently with `yazi`, and find it mildly frustrating to always have to confirm each deletion by typing `y<enter>`. I would rather configure `yazi` to default to `yes`, so I only have to type `<enter>` (of course that's my own risk and it's my fault if I delete files I didn't actually want to by accident, but I think the popup is already enough "friction" to not delete stuff my accident). If course, if you prefer it, you should defintely keep the `y/N` as the default in yazi, but configuring it to be `Y/n` would also be really nice!
Thanks!

**Comment by sxyazi** [comment]
Just `remove --force`, read our docs here: https://yazi-rs.github.io/docs/configuration/keymap#operation

If you want to achieve semi-automatic confirmation, simply bind it to two keys, such as `dy => remove --force` (`dy` means `delete and yes`), and `dd => remove` (confirmation required).

**Comment by mpr1255** [comment]
I love this software, really -- just jumped from nnn and have been amazed at how fast I've got up and running. But this advice is so cryptic, and parts of it are REALLY hard to configure. It's just not explained well. For example, here's my keymap.toml: 

```
[[input.prepend_keymap]]
on   = [ "<Esc>" ]
exec = "close"
desc = "Cancel input"


[[manager.prepend_keymap]]
on   = [ "l" ]
exec = "plugin --sync smart-enter"
desc = "Enter the child directory, or open the file"

[[manager.prepend_keymap]]
on   = [ "<Enter>" ]
exec = "plugin --sync smart-enter"
desc = "Enter the child directory, or open the file"

[[manager.append_keymap]]
on = [ "d" ]
exec = ["--force"]
desc = "Move the files to the trash" 
```

Well, that did nothing. Note that I tried prepend, append, the full command ("remove", "escape --visual --select" ) etc. etc.

I have no clue how to get this working. 

It's remarkable software; just needs better docs for a nicer onramp for new people. But it's better than nnn and it's brilliant.

**Reply by sxyazi** [reply]
I'm not quite sure I understand, is one of the functions not working - which one is it? Or are they both not working?

**Reply by sxyazi** [reply]
The obvious typo I can see is the last one, it should be ***(DANGER!! Pressing "d" will directly delete your files without any confirmation.)***:

```
[[manager.prepend_keymap]]
on = [ "d" ]
exec = ["remove --force"]
desc = "Move the files to the trash" 
```

Note using `prepend_keymap` instead of `append_keymap`, see the docs here https://yazi-rs.github.io/docs/configuration/keymap

**Reply by mpr1255** [reply]
Thank you. The last was not working, because I could not grok the syntax. The actual reason is because I am not very clever; but I blamed the software instead. 

It's working now and I love this thing. Insanely fast, so much functionality.... does 99% of what I'm asking of it and it's only like 3 months old. Incredible stuff.

**Comment by Vermoot** [comment]
Sorry to re-up this but it seems to me the behaviour asked by OP (which is what I'm trying to do as well) hasn't been reached?

Is there any way to keep the confirmation dialog, but have it be `Y/n` rather than `y/N`?

**Comment by sxyazi** [comment]
I plan to implement a new confirm component and it will allow users to customize their keybindings in the `keymap.toml`, should resolve this issue, https://github.com/sxyazi/yazi/issues/827#issuecomment-2007629604

**Comment by Vermoot** [comment]
That'll be great! Thanks

**Comment by sxyazi** [comment]
Done in https://github.com/sxyazi/yazi/pull/1167
